tests/fuzz/s2n_hybrid_ecdhe_kyber_r3_fuzz_test.c

```diff
@@ -55,27 +57,47 @@ static int setup_connection(struct s2n_connection *server_conn)
     server_conn->secure->cipher_suite = &s2n_ecdhe_kyber_rsa_with_aes_256_gcm_sha384;
     server_conn->handshake_params.conn_sig_scheme = s2n_rsa_pkcs1_sha384;
 
-    POSIX_GUARD(s2n_dup(&server_kem_params.private_key, &server_conn->kex_params.kem_params.private_key));
+    POSIX_GUARD(s2n_dup(&params->private_key, &server_conn->kex_params.kem_params.private_key));
     POSIX_GUARD(s2n_ecc_evp_generate_ephemeral_key(&server_conn->kex_params.server_ecc_evp_params));
 
     return S2N_SUCCESS;
 }
 
+int s2n_fuzz_init_kem_param(struct s2n_kem_params *param) {
+    POSIX_ENSURE_REF(param->kem);
+    struct s2n_blob *public_key_1 = &param->public_key;
+    POSIX_GUARD(s2n_alloc(public_key_1, S2N_KYBER_512_R3_PUBLIC_KEY_BYTES));
+
+    /* Do not GUARD this call to s2n_kem_generate_keypair(), as it will fail when attempting to generate the KEM
+     * key pair if !s2n_pq_is_enabled(). However, even if it does fail, it will have allocated zeroed memory for the
+     * public and private keys needed for setup_connection() to complete. */
+    s2n_result result = s2n_kem_generate_keypair(param);
+
+    if (s2n_pq_is_enabled()) {
+        POSIX_ENSURE_EQ(result.__error_signal, S2N_SUCCESS);
+    } else {
+        POSIX_ENSURE_NE(result.__error_signal, S2N_SUCCESS);
+    }
```

Comment:
You really shouldn't access result fields directly. Use s2n_result_is_ok or s2n_result_is_error

```suggestion
    POSIX_ENSURE_EQ(s2n_pq_is_enabled(), s2n_result_is_ok(result));
```